### PR TITLE
Subtask/as 2867 2869 lpa questionnaire magic link

### PR DESCRIPTION
**Pull request by adaMiclos**
## Ticket Number

https://pins-ds.atlassian.net/browse/AS-2867 
https://pins-ds.atlassian.net/browse/AS-2869

## Description of change

1. Create screens required for authentication.
2. Integrate service with magiclink-service-auth-api to authenticate the users.
3. Update all routes (except the ones used for authentication) to require authorisation.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/authentication.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+const mockEmail = 'test.email@test.uk';
+const mockLPACode = '123';
+
+const res = mockRes();
+const req = mockReq();
+req.params.lpaCode = mockLPACode;
+
+describe('authentication controller', () => {
+  describe('/:lpaCode/authentication/your-email', () => {
+    describe('show authentication enter email page', () => {
```
Comment (swdpcomputing) [inline packages/lpa-questionnaire-web-app/tests/unit/controllers/authentication.test.js:14]: If there's only one function in a describe, should it be so deeply nested?

Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/authentication.test.js:14]: In fact I think this whole file could be under one describe with the description under the 'it'. Open to having my mind changed if for a good reason.

File: packages/lpa-questionnaire-web-app/src/lib/magiclink-api-wrapper.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const fetch = require('node-fetch');
```
Comment (swdpcomputing) [inline packages/lpa-questionnaire-web-app/src/lib/magiclink-api-wrapper.js:1]: Looks good but can't see test file.

**Review by swdpcomputing — CHANGES_REQUESTED** [review]

File: packages/lpa-questionnaire-web-app/tests/unit/controllers/authentication.test.js (hunk 1 above)
Comment (adaMiclos) [reply packages/lpa-questionnaire-web-app/tests/unit/controllers/authentication.test.js]: I have removed one of the describe functions and updated the unit tests to be up to date with the latest code.

File: packages/lpa-questionnaire-web-app/tests/unit/middleware/authenticate.test.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+    res = mockRes();
+
+    jwtPayload = {
+      exp: 1912235086000,
+      userInformation,
+    };
+  });
+
+  describe('with expired jwtToken that has correct user information', () => {
```
[the remark this answers is not in the document]
Comment (adaMiclos) [reply packages/lpa-questionnaire-web-app/tests/unit/middleware/authenticate.test.js]: I've updated the test to have only  it statement.

File: packages/lpa-questionnaire-web-app/src/services/authentication/error/ExpiredJWTError.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module.exports = class ExpiredJWTError extends Error {
```
[the remark this answers is not in the document]
Comment (adaMiclos) [reply packages/lpa-questionnaire-web-app/src/services/authentication/error/ExpiredJWTError.js]: Test should be available now.

File: packages/lpa-questionnaire-web-app/src/lib/magiclink-api-wrapper.js (hunk 2 above)
Comment (adaMiclos) [reply packages/lpa-questionnaire-web-app/src/lib/magiclink-api-wrapper.js:1]: Test should be available now.

Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/src/lib/magiclink-api-wrapper.js:1]: Thank you.

File: packages/lpa-questionnaire-web-app/src/middleware/fetch-lpa.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const { getLPA } = require('../lib/appeals-api-wrapper');
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/src/middleware/fetch-lpa.js:1]: Thank you.

File: packages/lpa-questionnaire-web-app/src/services/authentication/authentication.service.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const JwtStrategy = require('passport-jwt/lib').Strategy;
```
[the remark this answers is not in the document]
Comment (swdpcomputing) [reply packages/lpa-questionnaire-web-app/src/services/authentication/authentication.service.js:1]: Thank you.

**Review by swdpcomputing — APPROVED** [review]

**Pull request merged**